src/plugins/chatview/utils.js

```diff
@@ -71,3 +73,117 @@ export function resetElementHeight (ev) {
         ev.target.style = '';
     }
 }
+
+
+/**
+ * Handle XEP-0147 "query actions" invoked via xmpp: URIs.
+ * Supports message sending, roster management, and future actions.
+ *
+ * Example URIs:
+ *   xmpp:user@example.com?action=message&body=Hello
+ *   xmpp:user@example.com?action=add-roster&name=John&group=Friends
+ */
+export async function routeToQueryAction(event) {
+    const { u } = _converse.env;
+
+    try {
```

Comment:
@marcellintacite  Why is it necessary to put everything in a `try/catch` here?

What happens when it's not there? Generally speaking, a try/catch should only be put around a single line or a small section of code that might be reasonably expected to throw an error that needs to be handled.